### PR TITLE
feat(OnyxInput): introduce `showPassword` property

### PR DESCRIPTION
**Pull request by bergser**
closes #4138 

Introduce `showPassword` property for the `OnyxInput` component

## Checklist

- [ ] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [ ] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) or functional test is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)
- [x] I have performed a self review of my code ("Files changed" tab in the pull request)
- [ ] All relevant changes are documented in the documentation app (folder `apps/docs`) if needed

### CODE REVIEW

File: packages/sit-onyx/src/components/OnyxInput/types.ts [hunk 1]
```diff
@@ -21,6 +21,10 @@ export type OnyxInputProps = SharedFormElementProps &
      * Whether to hide the clear icon when the input is filled and focused.
      */
     hideClearIcon?: boolean;
+    /**
+     * Show or hide the password value for input with type "password".
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxInput/types.ts]:
```suggestion
     * Show or hide the password value for input with `type="password"`.
     * If unset, the state will be managed internally.
     
```

File: packages/sit-onyx/src/components/OnyxInput/OnyxInput.vue [hunk 2]
```diff
@@ -94,9 +95,16 @@ const { disabled, showError } = useFormContext(props);
 const skeleton = useSkeletonContext(props);
 const errorClass = useErrorClass(showError);
 useAutofocus(input, props);
-const showPassword = ref(false);
+
+const showPassword = useVModel({
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxInput/OnyxInput.vue:103]: We also need to define an emit for the v-model:

```ts
const emit = defineEmits<{
  /**
   * Emitted when the password should be shown/hidden in plain text.
   */
  'update:showPassword': [value: boolean];
}>();
```

File: packages/sit-onyx/src/components/OnyxInput/OnyxInput.vue [hunk 3]
```diff
@@ -94,9 +95,16 @@ const { disabled, showError } = useFormContext(props);
...
+  props,
+  emit,
+  key: "showPassword",
+  default: false,
+});
+
 const displayType = computed(() => {
-  if (props.type === "password" && showPassword.value) {
+  if (props.type === "password" && props.showPassword) {
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxInput/OnyxInput.vue]:
```suggestion
  if (props.type === "password" && showPassword.value) {
```

We need to access the ref returned by "useVModel" instead of the prop so the value is correctly managed internally when the user does not provide a value :)

Also for the usages in line 193 and 194

File: packages/sit-onyx/src/components/OnyxInput/OnyxInput.vue [hunk 4]
```diff
@@ -30,6 +30,7 @@ const props = withDefaults(defineProps<OnyxInputProps>(), {
   loading: false,
   hideClearIcon: false,
   hideSuccessIcon: false,
+  showPassword: false,
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxInput/OnyxInput.vue]:
```suggestion
  showPassword: undefined,
```

When used with our `useVModel` utility, the default value here must be set to undefined so that the utility can differentiate whether the prop is not set by the user (undefined) or explicitly set (true or false)

File: .changeset/dirty-lamps-own.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"sit-onyx": minor
+---
+
+feat(OnyxInput): allow to manage `showPassword` state outside of the component
```
Comment (larsrickert) [inline .changeset/dirty-lamps-own.md]:
```suggestion
feat(OnyxInput): support new v-model `showPassword` to control the show password state when used with `type="password"`
```

File: packages/sit-onyx/src/components/OnyxInput/examples/PasswordVisibilityExample.vue
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxInput/examples/PasswordVisibilityExample.vue:1]: I don't think we need an example for this since its basically just using the v-model :)
Is that also fine for you?

**Comment by bergser** [comment]
> BTW I will let you be added to the SIT GitHub org so you can directly create branches in the onyx repo and don't have to work in the fork

thank you, but I think I'm already member of it ) just didn't know that I can create branch for myself in this repo.

**Comment by bergser** [comment]
@larsrickert comments are resolved. Not sure why `OnyxNavBar` tests are failing, in the PR there is no changes related to the nav bar 🤯

File: packages/sit-onyx/src/components/OnyxInput/OnyxInput.vue [hunk 6]
```diff
@@ -94,13 +99,22 @@ const { disabled, showError } = useFormContext(props);
...
+});
+
 const displayType = computed(() => {
   if (props.type === "password" && showPassword.value) {
     return "text";
   }
   return props.type;
 });
+
+watch(showPassword, (value) => emit("update:showPassword", value));
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxInput/OnyxInput.vue]:
```suggestion
```

The watch is not needed since the `useVModel` already calls it :)
So basically useVModel is similar to Vues defineModel but we needed to implement a custom utility because defineModel behaves differently in edge cases

**Comment by larsrickert** [comment]
> @larsrickert comments are resolved. Not sure why `OnyxNavBar` tests are failing, in the PR there is no changes related to the nav bar 🤯

Thanks! The nav bar tests might be flaky so re-running the checks should work

File: packages/sit-onyx/src/components/OnyxInput/types.ts [hunk 7]
```diff
@@ -21,6 +21,12 @@ export type OnyxInputProps = SharedFormElementProps &
      * Whether to hide the clear icon when the input is filled and focused.
      */
     hideClearIcon?: boolean;
+    /**
+     * Show or hide the password value for input with `type="password"`.
+     * If unset, the state will be managed internally.
+     
+     */
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxInput/types.ts]:
```suggestion
     */
```

**Review by larsrickert — APPROVED** [review]
